bbq/vm/context.go

```diff
@@ -55,7 +55,8 @@ type Context struct {
 	// This cache-alike is maintained per execution.
 	// TODO: Re-use the conversions from the compiler.
 	// TODO: Maybe extend/share this between executions.
-	semaTypeCache map[sema.TypeID]sema.Type
+	semaTypeCache   map[sema.TypeID]sema.Type
+	semaAccessCache map[interpreter.Authorization]sema.Access
```

Comment:
Is this cache only possible/used by the new generated subtype checking code, or is this a general optimization that could also be done on `master`? If the latter, it might be nice to port it over before, it's a great idea. That would keep this PR focused on just adding the generated subtype checking code.

Reply:
Ported https://github.com/onflow/cadence/pull/4336